### DISCUSSION TITLE
Task exited with return code Negsignal.SIGKILL

### DISCUSSION THREAD

**Discussion by parichay9** [apache/airflow#27742] [General]
### Apache Airflow version

2.4.3

### What happened

task running fine on local machine but getting error on production.
using astronomer:
airflow-2.4.3
pods: 28
cpu cores: 20
Memory: 60 GB
also while running the task memory is almost 45% of used.
I trying to get 500 rows per page for above 60 pages using api call then write to local json file then load to gcp bucket than big query

DAG(
        dag_id="xyz",
        schedule_interval="'36 9,16 * * *'",
        start_date=datetime(2022, 2, 17),
        default_args=default_args,
        max_active_runs=1,
        catchup=False
    )
PythonOperator(
                        task_id="abc",
                        python_callable=extraction,
                        op_kwargs={'config': endpoint_config, 'endpoint_extension': endpoint_extension},
                        provide_context=True
                    )

 task log screenshot:

<img width="1397" alt="Screenshot 2022-11-17 at 1 19 07 PM" src="https://user-images.githubusercontent.com/109953992/202444543-a91249c2-dd6c-449d-9735-e737d8e2a1c7.png">
scheduler log:

<img width="1098" alt="Screenshot 2022-11-17 at 1 15 43 PM" src="https://user-images.githubusercontent.com/109953992/202444280-ee192ab0-9b2f-4ce2-9156-62966cf04b28.png">

### Executor
Kubernetes

### Deployment

meteoric-kinetic-8423

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
There is no  way anyone would be able to do anything with the report without you going deeper, analysing long and explaining circumstatences when this happened. You need to explain when and how it happened, what you did when it happened, what you found when  analysing logs around the time it happened.

Otherwise you completely remove the possibility that someone will be able to help you in solving  the problem you have.
And in vast majority of such cases, this turns out to be some deployment misconfiguration. But  without more evidences, there is no way to know it.

Converted it into discussion.

**Reply by parichay9** [reply]
Thank you for suggestion i have attached information as much as i can.

**Reply by potiuk** [reply]
Still not enough info to help you. Some of your workers stopped responding - why how, hard to say - you need to see what it does. You also have not explained what deployment you have and which executor you have - but you need to  look at the tasks of yours to  see who and how kills it. Something (and it is not airflow - it's your code - likely crashes and gets killed - likely as a result of some serious problems - but there are not enough logs posted to see what happens. 

Then afterwards - airflow sees it and "reaps" the running tasks (zombie) when it realizes that the process had failed. 

But again - it's not possible to know what has failed and why - possibly you need to debug your code and try to pin-point who and when sends the signal.  It's really impossible to say without deeper investigation.
